packages/cloud/connection_details/types.ts

```diff
@@ -29,5 +30,5 @@ export interface ConnectionDetailsOptsApiKeys {
   createKey: (params: { name: string }) => Promise<{
     apiKey: ApiKey;
   }>;
-  hasPermission: () => Promise<boolean>;
+  hasPermission: () => Promise<ValidPermissionsResult | undefined>;
```

Comment:
The connection details flyout should not know anything about the Kibana security, can we leave it as `boolean`?